File: CodigoJava/CarreraMencion.java
package CodigoJava;
import java.util.Iterator;
import java.util.Objects;
import java.util.Stack;

public class CarreraMencion extends Carrera{
    private Stack<String> menciones;


    public CarreraMencion() {
    }

    public CarreraMencion(String nombre, int planEstudios, int idCarrera, Stack<String> menciones) {
        super(nombre, planEstudios, idCarrera);
        this.menciones = menciones;
    }
    
    public Stack<String> getA() {return this.menciones;}
    public void setA(Stack<String> menciones) {this.menciones = menciones;}

    public CarreraMencion a(Stack<String> a) {
        setA(a);
        return this;
    }
    public void leer(){
        super.leer();
        System.out.println("- Lectura Menciones");
        char condition=' ';
		do {
            System.out.print("\t>> Inserte mencion => ");
            String men=Leer.dato();
            menciones.add(men);
			System.out.println(">>>>>> Desea seguir añadiendo menciones? (s/n) => ");
			String g =Leer.dato();
			condition= g.charAt(0);
		} while (condition=='s');
        
        
    } 

    public void mostrarMenciones(){
        Iterator<String> r=menciones.iterator();
        while(r.hasNext()){
            System.out.println("\t-"+r.next());
        }
    } 

    @Override
    public boolean equals(Object o) {
        if (o == this)
            return true;
        if (!(o instanceof CarreraMencion)) {
            return false;
        }
        CarreraMencion carreraMencion = (CarreraMencion) o;
        return Objects.equals(menciones, carreraMencion.menciones);
    }




}

File: c#/Interfaz Grafica/Guia_Centros_Estudio_Superior.Core/Models/CarreraMencion.cs
using System;
using System.Collections;
using System.Linq;
using System.Threading.Tasks;


using System.IO;
namespace Guia_Centros_Estudio_Superior.Core.Models;
[Serializable]

public class CarreraMencion: Carrera
{
    private Stack<string> menciones;

    public Stack<string> Menciones
    {
        get => menciones;
        set => menciones = value;
    }

    public CarreraMencion() {
        Menciones=new Stack<string>();
    }

    public CarreraMencion(string nombre, int planEstudios, int idCarrera, Stack<string> menciones): base(nombre, planEstudios, idCarrera) {
        this.Menciones = menciones;
    }
    
    public Stack<string> getA() {return this.Menciones;}
    public void setA(Stack<string> menciones) {this.Menciones = menciones;}

    public CarreraMencion a(Stack<string> a) {
        setA(a);
        return this;
    }
    public new void leer(){
        base.leer();
        Console.WriteLine("- Lectura Menciones");
		do {
            Console.Write("\t>> Inserte mencion => ");
            string men=Leer.Cadena;
            Menciones.Append(men);
			Console.WriteLine(">>>>>> Desea seguir añadiendo menciones? (s/n) => ");
    	}while(Console.ReadKey().KeyChar=='s');
    } 

    public void mostrarMenciones(){
        IEnumerator<string> r=Menciones.GetEnumerator();
        while(r.MoveNext()){
            Console.WriteLine("\t-"+r.Current);
        }
    } 

    //@Override
    /*
    public bool equals(Object o) {
        if (o == this)
            return true;
        if (!(o instanceof CarreraMencion)) {
            return false;
        }
        CarreraMencion carreraMencion = (CarreraMencion) o;
        return Objects.equals(menciones, carreraMencion.menciones);
    }
    */

        
}
